src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/close_application_request_test.cc

```diff
@@ -0,0 +1,345 @@
+  using namespace am;
+  auto message = std::make_shared<SmartObject>(smart_objects::SmartType_Null);
+  auto& params = (*message)[strings::params];
+  params[strings::connection_key] = kConnectionKey;
+  params[strings::function_id] = mobile_apis::FunctionID::CloseApplicationID;
+  return message;
+}
+
+class CloseApplicationRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  CloseApplicationRequestTest()
+      : mock_app_(CreateMockApp())
+      , message_(CreateCloseAppMessage())
+      , command_(
+            CreateCommand<sdl_rpc_plugin::commands::CloseApplicationRequest>(
+                message_)) {}
+
+ protected:
+  void SetUp() OVERRIDE {
+    using namespace am;
+    ON_CALL(app_mngr_, application(kConnectionKey))
+        .WillByDefault(Return(mock_app_));
+    ON_CALL(app_mngr_, GetRPCService())
+        .WillByDefault(ReturnRef(mock_rpc_service_));
+    ON_CALL(app_mngr_, GetNextHMICorrelationID())
+        .WillByDefault(Return(kCorrelationId + 1));
+    ON_CALL(app_mngr_, GetPolicyHandler())
+        .WillByDefault(ReturnRef(mock_policy_handler_));
+    ON_CALL(app_mngr_, connection_handler())
+        .WillByDefault(ReturnRef(mock_connection_handler_));
+    ON_CALL(mock_connection_handler_, get_session_observer())
+        .WillByDefault(ReturnRef(mock_session_handler_));
+    ON_CALL(*mock_app_, app_id()).WillByDefault(Return(kConnectionKey));
+  }
+
+  MockAppPtr mock_app_;
+  MessageSharedPtr message_;
+  NiceMock<components::connection_handler_test::MockConnectionHandler>
+      mock_connection_handler_;
+  components::protocol_handler_test::MockSessionObserver mock_session_handler_;
+  CloseApplicationCommandPtr command_;
+};
+
+TEST_F(CloseApplicationRequestTest,
+       CloseApplicationRequestSendsResponse_APPLICATION_NOT_REGISTERED) {
+  using namespace am;
+  auto message = CreateCloseAppMessage();
+
+  CloseApplicationCommandPtr command(
+      CreateCommand<sdl_rpc_plugin::commands::CloseApplicationRequest>(
+          message));
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(MockAppPtr()));
+
+  EXPECT_CALL(mock_rpc_service_, ManageHMICommand(_)).Times(0);
+
+  auto response_message = CreateMessage();
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageMobileCommand(_, am::commands::Command::CommandSource::SOURCE_SDL))
+      .WillOnce(DoAll(SaveArg<0>(&response_message), Return(true)));
+  command->Run();
+
+  const auto result_code = static_cast<mobile_apis::Result::eType>(
+      (*response_message)[strings::msg_params][strings::result_code].asInt());
+  EXPECT_EQ(result_code, mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+
+  const bool success =
+      (*response_message)[strings::msg_params][strings::success].asBool();
+  EXPECT_FALSE(success);
+}
+
+TEST_F(CloseApplicationRequestTest,
+       CloseApplicationRequestSendsResponse_IGNORED) {
+  using namespace am;
+  auto message = CreateCloseAppMessage();
```

Comment:
@IGapchuk We can move creation `message` and `response_message` to SetUp method I think.

Reply:
@ZhdanovP Fixed in a465d3a1e8eb9abeaeb8836a4917d23c59be3648